pkg/ottl/parser.go

```diff
@@ -158,11 +156,30 @@ func newParser[G any]() *participle.Parser[G] {
 
 // Statements represents a list of statements that will be executed sequentially for a TransformContext.
 type Statements[K any] struct {
-	statements        []Statement[K]
+	statements        []*Statement[K]
 	errorMode         ErrorMode
 	telemetrySettings component.TelemetrySettings
 }
 
+type StatementsOption[K any] func(*Statements[K])
+
+func WithErrorMode[K any](errorMode ErrorMode) StatementsOption[K] {
+	return func(s *Statements[K]) {
+		s.errorMode = errorMode
+	}
+}
+
+func NewStatements[K any](statements []*Statement[K], telemetrySettings component.TelemetrySettings, options ...StatementsOption[K]) Statements[K] {
```

Comment:
Ya its a little ambiguous.  The reason to use the Statements struct is to get access to the `Execute` (and future `Eval`) functions and their value.  If we can come up with a good name that conveys how this wrapper is useful that would be good.

Reply:
@evan-bradley I opened https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/18636 to track this name.  I think we can move forward with this PR for now so that we can add the configuration options, which will be independent of any rename.

Reply:
Agreed, we should do that separately from this PR. Thanks for opening that.